src/main/java/esme/ui/Ui.java

```diff
@@ -0,0 +1,274 @@
+     */
+    public void handleTaskStatus(String[] words) {
+        String command = words[0];
+        try {
+            if (taskList.getNumberOfTasks() <= 0) {
+                throw new EsmeException("Add some tasks! Currently, there are no tasks to be " +
+                        command + "ed.");
+            }
+            if (words.length != 2) {
+                throw new EsmeException("Error: Wrong format! Please use the format: command index (e.g., '" +
+                        command + " 1')");
+            }
+            int index = generateIndex(words[1]);
+            if (!isIndexValid(index)) {
+                throw new EsmeException("Oh dear, it seems the index has wandered beyond the " +
+                        "boundaries of our list!");
+            }
+            if (!isTaskCompleted(index) && command.equals("unmark")) {
+                throw new EsmeException("The stars have revealed that this task is yet to be completed. " +
+                        "Finish it before the cosmos frowns upon you.");
+            }
+            if (isTaskCompleted(index) && command.equals("mark")) {
+                throw new EsmeException("This task has already been blessed by the cosmos. " +
+                        "Continue your journey with the next task.");
+            }
+            toggleTaskStatus(index,command);
+        } catch (EsmeException e) {
+            displayLine(true);
+            System.out.println("\t" + e.getMessage());
+            displayLine(true);
+        }
+    }
+
+    public void toggleTaskStatus(int taskIndex, String command) {
+        if (command.equals("mark")) {
+            markTaskInList(taskIndex);
+        } else {
+            unmarkTaskInList(taskIndex);
+        }
+    }
+
+    public void printTaskList() {
+        taskList.printTaskList();
+        displayLine(true);
+    }
+
+
+    /**
+     * Prints a line to the console to separate different sections of the UI.
+     * If hasIndent is true, the line is indented with 4 spaces.
+     * 
+     * @param hasIndent Whether the line should be indented or not.
+     */
+    public void displayLine(boolean hasIndent) {
+        if (hasIndent) {
+            System.out.println("\t" + SEPARATOR);
+        } else {
+            System.out.println(SEPARATOR);
+        }
+    }
+
+    public void greet() {
+        displayLine(false);
+        System.out.println("Hello! I'm");
+        System.out.print(esmeLogo);
+        System.out.println(", the Astrologer. The cosmos whispers its secrets to me.");
+        displayLine(false);
+        System.out.println("How may I assist you today? The stars and I are at your service.");
+        displayLine(false);
+    }
+
+    public void farewell() {
+        displayLine(true);
+        System.out.println("\tAu revoir, mon ami! May the cosmos continue to weave a tapestry of fortune" +
+                " in your favor!");
+        displayLine(true);
```

Comment:
You might want to extract this code pattern further into a separate method and pass in parameters as the same pattern is used in multiple methods.